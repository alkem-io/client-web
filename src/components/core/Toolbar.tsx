import clsx from 'clsx';
import React, { FC, RefObject } from 'react';
import { createStyles } from '../../hooks';

const useToolbarStyles = createStyles(theme => ({
  toolbar: {
    display: 'flex',
    flexDirection: 'row',
  },
  paddingDefault: {
    padding: `${theme.shape.spacing(4)}px ${theme.shape.spacing(4)}px`,
  },
  paddingDense: {
    padding: `${theme.shape.spacing(2)}px ${theme.shape.spacing(4)}px`,
  },
  responsivePadding: {
    [theme.media.down('md')]: {
      paddingLeft: theme.shape.spacing(2),
      paddingRight: theme.shape.spacing(2),
    },
  },
}));

interface ToolbarProps {
  dense?: boolean;
  classes?: {
    padding?: string;
    densePadding?: string;
  };
  className?: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  innerRef?: RefObject<any>;
}

const Toolbar: FC<ToolbarProps> = ({ dense, classes, className, children, innerRef }) => {
  const styles = useToolbarStyles();

  return (
    <div
      ref={innerRef}
      className={clsx(
        styles.toolbar,
        classes?.padding || styles.paddingDefault,
        dense && (classes?.densePadding || styles.paddingDense),
        styles.responsivePadding,
        className
      )}
    >
      {children}
    </div>
  );
};

export default Toolbar;
