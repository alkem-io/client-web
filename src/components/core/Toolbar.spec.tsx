import { cleanup, render } from '../../utils/test/test-utils';
import Toolbar from './Toolbar';

afterEach(cleanup);
describe('Toolbar - main', () => {
  test('render correctly Toolbar component', () => {
    // arrange
    const { asFragment } = render(<Toolbar />);

    // act
    const html = asFragment();

    // assert
    expect(html).toMatchSnapshot();
  });
});
