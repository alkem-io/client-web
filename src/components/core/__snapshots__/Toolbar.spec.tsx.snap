// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar - main render correctly Toolbar component 1`] = `
<DocumentFragment>
  <div
    class="toolbar-0-2-1 paddingDefault-0-2-2 responsivePadding-0-2-4"
  />
</DocumentFragment>
`;
